### PR TITLE
Unit tests for UrlSigner

### PR DESCRIPTION
**Pull request by evildour**

### CODE REVIEW

File: apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.Tests/UrlSignerTest.cs [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        }
+
+        [Fact]
+        public void Sign_Validations()
+        {
+            var signer = UrlSigner.FromServiceAccountCredential(
+                GoogleCredential.GetApplicationDefaultAsync().Result.UnderlyingCredential as ServiceAccountCredential);
+
+            Assert.Throws<ArgumentNullException>(() => signer.Sign(null, "objectName", TimeSpan.FromDays(1), new HttpRequestMessage()));
+            Assert.Throws<ArgumentException>(() => signer.Sign("BUCKETNAME", "objectName", TimeSpan.FromDays(1), new HttpRequestMessage()));
```
Comment (jskeet) [inline apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.Tests/UrlSignerTest.cs]: Possibly add a comment to explain why this is expected to fail?

File: apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.Tests/UrlSignerTest.cs [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        [Fact]
+        public void EncryptionKeyAndHashAreIgnored()
+        {
+            var signer = UrlSigner.FromServiceAccountCredential(
+                GoogleCredential.GetApplicationDefaultAsync().Result.UnderlyingCredential as ServiceAccountCredential);
+
+            var bucketName = "bucket-name";
+            var objectName = "object-name";
+            var expiration = DateTimeOffset.UtcNow + TimeSpan.FromDays(1);
+            var url1 = signer.Sign(bucketName, objectName, expiration, requestHeaders: new Dictionary<string, IEnumerable<string>> {
```
Comment (jskeet) [inline apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.Tests/UrlSignerTest.cs]: I'm pretty sure the unit test project has internal access to the EncryptionKey constants...

**Comment by jskeet** [comment]
(Obviously need to get the tests working first :)

File: apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.Tests/UrlSignerTest.cs (hunk 1 above)
Comment (evildour) [inline apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.Tests/UrlSignerTest.cs]: Done

File: apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.Tests/UrlSignerTest.cs (hunk 2 above)
Comment (evildour) [inline apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.Tests/UrlSignerTest.cs]: Done

**Comment by evildour** [comment]
Yup, forgot that the CI server won't have default credentials.

**Comment by evildour** [comment]
PTAL

File: apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.Tests/UrlSignerTest.cs [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+        [Fact]
+        public void FromServiceAccountCredential_Validations()
+        {
+            Assert.Throws<ArgumentNullException>(() => UrlSigner.FromServiceAccountCredential(null));
+        }
+
+        [Fact]
+        public void Sign_Validations()
+        {
+            var signer = UrlSigner.FromServiceAccountCredential(new ServiceAccountCredential(new ServiceAccountCredential.Initializer("test")
```
Comment (jskeet) [inline apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.Tests/UrlSignerTest.cs]: Possibly extract this to a separate method, given that we've got it in a few places?

Comment (evildour) [inline apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.Tests/UrlSignerTest.cs]: Done

**Comment by evildour** [comment]
Hopefully the tests should work now. Will merge when green.

**Comment by evildour** [comment]
@jskeet have you seen the appveyor error before? It looks like these unit tests ran and passed before that error, so it should be good to merge, but I just want to double check.

**Comment by jskeet** [comment]
Haven't seen that before - rebuilding now.

**Comment by evildour** [comment]
Still failing. Given the timing and since it keeps failing after fetching protobuf, I'm wondering if it has to do with the merge of [this PR](https://github.com/google/protobuf/pull/2701).

**Comment by jskeet** [comment]
Ah, yes, that would be it - I need to adjust the build scripts. Yes, feel free to ignore that, and I'll fix it tomorrow.

**Pull request merged**